Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/lists/HasStableId.kt

```diff
@@ -0,0 +1,5 @@
+package de.rki.coronawarnapp.util.lists
+
+interface HasStableId {
```

Comment:
Isn't that somewhat misused? E.g. the Comparable interface is implemented if a class fulfills that criteria, otherwise it does not implement this.

Reply:
>  E.g. the Comparable interface is implemented if a class fulfills that criteria, otherwise it is not.

Isn't that what we are doing here? An adapter class can implement this interface to signal that it provides a `stableId`. So for all classes that offer this "feature" by implementing this interface, we can use classes on the adapter that don't need to know anything but the stable ID. i.e. `AsyncDiffUtilAdapter`.